chirho/dynamical/handlers/trajectory.py

```diff
@@ -27,6 +31,19 @@ def __enter__(self) -> "LogTrajectory[T]":
         self.trajectory: State[T] = State()
```

Comment:
@eb8680 , do you think it makes sense to move this line into `_pyro_simulate` instead of `__enter__`. This would change the behavior from "concatenate multiple `simulate` calls trajectories together" to "store only the trajectory of the final simulate call". Or alternatively, we could give `simulate` and optional name argument if you want to store multiple trajectories.

Reply:
Yes, re-initializing `self.trajectory` in `_pyro_simulate` seems more correct than the current behavior.